bentoml/_internal/bento/build_config.py

```diff
@@ -554,9 +635,11 @@ class BentoBuildConfig:
     )
 
     def __attrs_post_init__(self) -> None:
-        use_conda = any(
-            val is not None for _, val in bentoml_cattr.unstructure(self.conda).items()
-        )
+        use_conda = False
+        for attr_ in [a.name for a in attr.fields(self.conda.__class__)]:
+            if getattr(self.conda, attr_) is not None:
+                use_conda = True
+                break
```

Comment:
move this to conda options:

```
   def is_used(self):
          return not self == CondaOptions()
```

`is not None` may not be the right check here, it should verify if it's the default value. it could break if a new option being added has a default value of an empty list.

Reply:
seems like we can do `__eq__` for all of the options class?